src/Compilers/Core/Portable/Symbols/ISymbolExtensions_IsAccessibleWithin.cs

```diff
@@ -0,0 +1,353 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    public static partial class ISymbolExtensions
+    {
+        /// <summary>
+        /// Checks if 'symbol' is accessible from within 'within', with
+        /// an optional qualifier of type 'throughTypeOpt' to be used to resolve
+        /// protected access.
+        /// </summary>
+        public static bool IsAccessibleWithin(
+            this ISymbol symbol,
+            ISymbol within,
+            ITypeSymbol throughTypeOpt = null)
+        {
+            if (symbol == null)
+            {
+                throw new ArgumentNullException(nameof(symbol));
+            }
+
+            if (within == null)
+            {
+                throw new ArgumentNullException(nameof(within));
+            }
+
+            if (!(within is INamedTypeSymbol || within is IAssemblySymbol))
+            {
+                throw new ArgumentException("must be an INamedTypeSymbol or an IAssemblySymbol", nameof(within));
```

Comment:
since htis is a public API, i think these need to be loc'ed resources.  Right?

Reply:
We have mixed precedence. I'll localize.

---
In reply to: [184568430](https://github.com/dotnet/roslyn/pull/26367#discussion_r184568430) [](ancestors = 184568430)